src/text-cairo.c

```diff
@@ -1099,6 +1121,7 @@ cairo_MeasureCharacterRanges (GpGraphics *graphics, GDIPCONST WCHAR *stringUnico
 	RectF			charRect;
 	RectF 			rc_coords, *layoutRect = &rc_coords;
 	BOOL			optimize_convert;
+	int			index_matching[length];
```

Comment:
This is a gnu extension - I think you need to change this to

```
int* index_matching = (int *) malloc (sizeof (int) * length));
if (!index_matching)
    return OutOfMemory;
```

Also should be freed in `cleanup:`

Reply:
Actually it's C99, but agreed with the allocation.